rustls/src/server/mod.rs

```diff
@@ -371,10 +372,11 @@ impl ServerSessionImpl {
             state: Some(Box::new(hs::ExpectClientHello::new(
                 server_config,
                 extra_exts,
+                SessionRandoms::for_server()?,
```

Comment:
IIUC, the API breaking change here is due to the refactoring where the randoms are generated early, during construction of the session, right? IMO, this is the second strike against this idea. It seems better to delay the generation of the randoms until they are needed. Since those code paths are already fallable to deal with I/O (IIUC), then no API change would be needed.

Reply:
Fair enough, I'll look into that next.

Reply:
This has been solved (for the server side).